### ISSUE TITLE
podman machine start fails due to zero machine config

### ISSUE THREAD

**Issue by afbjorklund** [containers/podman#11824]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

Upgrading the podman configuration, hangs the machine boot.

**Steps to reproduce the issue:**

1. `podman machine init`

2.  `podman machine start`

3.

**Describe the results you received:**

Hangs, `Waiting for VM ....` There is no feedback.

When running with debug logs, the GUI show that it fails to load the initramfs.

![machine-start](https://user-images.githubusercontent.com/10364051/135612278-5c7a00a5-c757-4ac9-9342-a43a4a6e0492.png)

The root cause is that the VM gets no cpu and no ram, from the qemu command.

`[/usr/bin/qemu-system-x86_64 -m 0 -smp 0 ...
`
There should probably be some validation, at least one CPU and at least one GB ?

**Describe the results you expected:**

The VM should start up.

**Additional information you deem important (e.g. issue happens only occasionally):**

This is the configuration, written to containers.conf

```
[machine]
  cpus = 0
  disk_size = 0
  memory = 0

```

**Output of `podman version`:**

```
podman version 4.0.0-dev
```

**Output of `podman info --debug`:**

```
(paste your output here)
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
(paste your output here)
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/master/troubleshooting.md)**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Ubuntu 20.04

**Comment by afbjorklund** [comment]
Also, after running podman4 machine, podman3 machine shows warnings. But I guess that is to be expected... ?

`WARN[0000] Failed to decode the keys ["engine.service_timeout" "machine" "machine.cpus" "machine.disk_size" "machine.memory"] from "/home/anders/.config/containers/containers.conf". `

**Comment by Luap99** [comment]
I think the problem is that toml writes the empty values to the config file. Instead of `omitempty` we have to use `omitzero` for numeric config fields, see https://github.com/BurntSushi/toml/issues/120

This is a serious problem because we already released this with podman 3.4. Once this is written to the config file every new VM will be broken.

**Comment by afbjorklund** [comment]
The workaround is to supply all the parameters to the command.

`podman machine init --cpus 2 --memory 2048 --disk-size 10`

**Comment by Luap99** [comment]
Well you can also remove the bad config stuff
```
[machine]
  cpus = 0
  disk_size = 0
  memory = 0
```
It should work for the next `podman machine init` but after this the config file will be broken again.
PR https://github.com/containers/common/pull/795 should fix this so that we do not write zero values to the config.

**Comment by Luap99** [comment]
Reopen because the fix is not in podman yet.

**Comment by Luap99** [comment]
This is fixed in the 3.4 branch and main so I close this.
